src/pallene/pallenelib.lua

```diff
@@ -43,9 +43,73 @@ return [==[
 #include <stdarg.h>
 #include <locale.h>
 #include <math.h>
+#include <stdbool.h>
 
 #define PALLENE_UNREACHABLE __builtin_unreachable()
 
+/* Part of Pallene Tracer. */
+#define PALLENE_C_FRAMEENTER(L, name)   pt_fn_details_t _details = {                             \
+                                            .fn_name  = name,                                    \
+                                            .mod_name = PALLENE_SOURCE_FILE                      \
+                                        };                                                       \
```

Comment:
Make sure that the _details are statically allocated. `static pt_fn_details_t _details` ought to do the trick.

Reply:
But why tho?

Reply:
Should be more efficient. Only one copy of the _details struct, instead of making a new one for every stack frame.